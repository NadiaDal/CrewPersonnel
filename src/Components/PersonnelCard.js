// @flow

import React from 'react';
import { all, isEmpty, keys, pipe } from 'ramda';

export type PersonnelType = {
  id: number,
  name: string,
  email: string,
  picture: string,
  location: string,
  category: string,
};

export type PersonnelCardType = {
  search: Object,
  person: PersonnelType,
  children: any,
};

const isSubStringByProp = ({
  prop,
  item,
  search,
}: {
  prop: string,
  item: PersonnelType,
  search: { name: string, location: string },
}) => {
  if (isEmpty(search[prop])) return true;
  return item[prop].toLowerCase().indexOf(search[prop].toLowerCase()) !== -1;
};

export const isVisibleBySearch = (item: PersonnelType, search: { name: string, location: string }) =>
  pipe(
    keys,
    all(prop => isSubStringByProp({ prop, item, search })),
  )(search);

const PersonnelCard = ({ person, search, children }: PersonnelCardType) => {
  const { name, email, picture, location } = person;
  const isVisible = isVisibleBySearch(person, search);

  return isVisible ? (
    <div style={styles.container}>
      <div style={styles.image}>
        <img src={picture} alt="" />
      </div>
      <div style={styles.name}>{name}</div>
      <div style={styles.text}>{location}</div>
      <div style={styles.text}>{email}</div>
      <div style={styles.buttonContainer}>{children}</div>
    </div>
  ) : null;
};

const styles = {
  container: {
    backgroundColor: '#f1f1f1',
    padding: '10px',
    margin: '10px',
    wordBreak: 'break-all',
  },
  image: {
    width: 60,
    height: 60,
    borderRadius: '50%',
    overflow: 'hidden',
  },
  name: {},
  text: {},
  buttonContainer: {
    display: 'flex',
    justifyContent: 'space-between',
  },
};

export default PersonnelCard;
