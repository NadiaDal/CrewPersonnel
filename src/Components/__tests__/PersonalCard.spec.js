import React from 'react';
import renderer from 'react-test-renderer';
import PersonnelCard from '../PersonnelCard';

const person = {
  id: 0,
  category: 'applied',
  email: 'brayden.hamilton@example.com',
  location: 'Nottingham',
  name: 'Mr Brayden Hamilton',
  picture: 'https://randomuser.me/api/portraits/med/men/91.jpg',
};

it('personnel card with empty search fields renders correctly', () => {
  const search = { name: '', location: '' };

  const tree = renderer.create(<PersonnelCard search={search} person={person} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('personnel card match by name renders correctly', () => {
  const search = { name: 'Bra', location: '' };

  const tree = renderer.create(<PersonnelCard search={search} person={person} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('personnel card match by location renders correctly', () => {
  const search = { name: '', location: 'Nottin' };

  const tree = renderer.create(<PersonnelCard search={search} person={person} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('personnel card does not match by name renders correctly', () => {
  const search = { name: 'Mag', location: 'Nottin' };

  const tree = renderer.create(<PersonnelCard search={search} person={person} />).toJSON();

  expect(tree).toBeNull();
});

it('personnel card does not match by location renders correctly', () => {
  const search = { name: 'Hamilton', location: 'Coventry' };

  const tree = renderer.create(<PersonnelCard search={search} person={person} />).toJSON();

  expect(tree).toBeNull();
});
