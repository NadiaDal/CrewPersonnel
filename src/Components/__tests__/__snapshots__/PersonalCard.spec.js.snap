// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`personnel card match by location renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgba(241, 241, 241, 0.8)",
      "borderRadius": "7px",
      "margin": "10px",
      "padding": "10px 15px",
      "wordBreak": "break-all",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "marginRight": "20px",
        }
      }
    >
      <img
        alt=""
        src="https://randomuser.me/api/portraits/med/men/91.jpg"
        style={
          Object {
            "borderRadius": "50%",
            "height": 60,
            "width": 60,
          }
        }
      />
    </div>
    <div>
      <div
        style={
          Object {
            "fontSize": "16px",
            "marginBottom": "10px",
          }
        }
      >
        Mr Brayden Hamilton
      </div>
      <div
        style={
          Object {
            "fontSize": "14px",
          }
        }
      >
        Nottingham
      </div>
      <div
        style={
          Object {
            "fontSize": "14px",
          }
        }
      >
        brayden.hamilton@example.com
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  />
</div>
`;

exports[`personnel card match by name renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgba(241, 241, 241, 0.8)",
      "borderRadius": "7px",
      "margin": "10px",
      "padding": "10px 15px",
      "wordBreak": "break-all",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "marginRight": "20px",
        }
      }
    >
      <img
        alt=""
        src="https://randomuser.me/api/portraits/med/men/91.jpg"
        style={
          Object {
            "borderRadius": "50%",
            "height": 60,
            "width": 60,
          }
        }
      />
    </div>
    <div>
      <div
        style={
          Object {
            "fontSize": "16px",
            "marginBottom": "10px",
          }
        }
      >
        Mr Brayden Hamilton
      </div>
      <div
        style={
          Object {
            "fontSize": "14px",
          }
        }
      >
        Nottingham
      </div>
      <div
        style={
          Object {
            "fontSize": "14px",
          }
        }
      >
        brayden.hamilton@example.com
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  />
</div>
`;

exports[`personnel card with empty search fields renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgba(241, 241, 241, 0.8)",
      "borderRadius": "7px",
      "margin": "10px",
      "padding": "10px 15px",
      "wordBreak": "break-all",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "marginRight": "20px",
        }
      }
    >
      <img
        alt=""
        src="https://randomuser.me/api/portraits/med/men/91.jpg"
        style={
          Object {
            "borderRadius": "50%",
            "height": 60,
            "width": 60,
          }
        }
      />
    </div>
    <div>
      <div
        style={
          Object {
            "fontSize": "16px",
            "marginBottom": "10px",
          }
        }
      >
        Mr Brayden Hamilton
      </div>
      <div
        style={
          Object {
            "fontSize": "14px",
          }
        }
      >
        Nottingham
      </div>
      <div
        style={
          Object {
            "fontSize": "14px",
          }
        }
      >
        brayden.hamilton@example.com
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  />
</div>
`;
